content/200-concepts/100-components/01-prisma-schema/06-relations/07-referential-actions.mdx

```diff
@@ -8,6 +8,401 @@ preview: true
+```prisma file=schema.prisma highlight=4,5;add
+model Post {
+  id          Int     @id @default(autoincrement())
+  title       String
+  authorName  String  @default("anonymous")
+  author      User    @relation(fields: [authorId], references: [id], onDelete: SetDefault, onUpdate: SetDefault)
+  authorId    Int
+}
+
+model User {
+  id        Int    @id @default(autoincrement())
+  username  String @id
+  posts     Post[]
+}
+```
+
+##### Result of using `SetDefault`?
+
+When deleting a `User`, its existing authored posts will now be authored by a `User` with `id` "anonymous". 
+
+When the `username` of a `User` changes, its existing authored posts will now be authored by `User` with `id` "anonymous".
+
+## Behavior in <=2.25.0 or without Preview flag <inlinecode>referentialActions</inlinecode>
+
+<a id="current"></a>
+
+The following behavior applies in 2.25.0 and earlier, and since 2.26.0 _without_ the `referentialActions` preview flag.
+
+When invoking the [`delete()`](../../../../guides/database/advanced-database-tasks/cascading-deletes/postgresql#91-validating-restrict) or [`deleteAll()`](../../prisma-client/crud#delete-all-records) methods using the Prisma Client, a runtime check is performed and the deletion of records prevented if they are referencing related objects. **This prevents cascade behavior**.
+
+The current behavior, without upgrading and enabling the feature flag, does not allow setting referential actions at all. 
+
+### The <=2.25.0 default referential actions
+
+| Clause     | Optional relations | Mandatory relations |
+| :--------- | :----------------- | :------------------ |
+| `onDelete` | `SetNull`          | `Cascade`           |
+| `onUpdate` | `Cascade`          | `Cascade`           |
+
+If you need to actually use the cascade behavior configured in the database, you _can_ use [`$raw`](../../prisma-client/raw-database-access) SQL queries to [delete multiple referenced records](../../prisma-client/crud#deleting-all-data-with-raw-sql--truncate) (such as users blog posts) and the deletion will be successful. This is because the Prisma Client will **not** perform runtime checks on raw queries.
+
+## Upgrade paths
+
+There are a couple of paths you can take when upgrading which will give different results depending on the desired outcome.
+
+The following assumes you have upgraded to 2.26.0 and [enabled the preview feature flag](#enable-the-referentialactions-preview-feature).
+
+### Using Introspection
+
+If you [Introspect](../introspection) your database **before modifying your data model**, the [<=2.25.0 default values](#the-2250-default-referential-actions) will be reflected in your prisma schema. 
+
+Of note is the default value of `onDelete`, which <=2.25.0 is `Cascade`. 
+
+<Admonition type="info">
+
+The [Prisma VSCode extension](https://marketplace.visualstudio.com/items?itemName=Prisma.prisma) (if installed) will automatically add `onDelete: Cascade` to any required relations defined in your prisma schema.
```

Comment:
I wasn't aware of this. Prisma Introspect makes a lot of sense here, but I am surprised to see the VS code extension. Does it do that, really?

Reply:
Yep! Its cool. You add the preview feature flag then introspect and the extension adds `onDelete: Cascade` to any required relations. I guess the thinking is to highlight that its not using the new default?

Reply:
Solved in slack